src/AutoRest.CSharp/Common/Output/Models/Types/HelperTypeProviders/MultipartFormDataRequestContentProvider.cs

```diff
@@ -0,0 +1,495 @@
+                Assign(ContentDispositionHeaderExpression, header),
+            };
+            return new Method(signature, body);
+        }
+
+        private Method BuildAddContentTypeHeaderMethod()
+        {
+            var contentParam = new Parameter("content", null, typeof(HttpContent), null, ValidationType.None, null);
+            var contentTypeParam = new Parameter("contentType", null, typeof(string), null, ValidationType.None, null);
+            var signature = new MethodSignature(
+                Name: _addContentTypeHeaderMethodName,
+                Summary: null,
+                Description: null,
+                Modifiers: MethodSignatureModifiers.Public | MethodSignatureModifiers.Static,
+                ReturnType: null,
+                ReturnDescription: null,
+                Parameters: new[]
+                {
+                    contentParam,
+                    contentTypeParam,
+                });
+            var contentTypeExpression = (ValueExpression)contentTypeParam;
+            ValueExpression ContentTypeHeaderExpression = ((ValueExpression)contentParam).Property("Headers").Property("ContentType");
+            var body = new MethodBodyStatement[]
+            {
+                Declare(typeof(MediaTypeHeaderValue), "header", New.Instance(typeof(MediaTypeHeaderValue), new[]{ contentTypeExpression }), out var header),
+                Assign(ContentTypeHeaderExpression, header),
+            };
+            return new Method(signature, body);
+        }
+
+        private Method BuildTryComputeLengthMethod()
+        {
+            var lengthParameter = new Parameter("length", null, typeof(long), null, ValidationType.None, null, IsOut: true);
+            var signature = new MethodSignature(
+                Name: "TryComputeLength",
+                Summary: null,
+                Description: null,
+                Modifiers: MethodSignatureModifiers.Public | MethodSignatureModifiers.Override,
+                ReturnType: typeof(bool),
+                ReturnDescription: null,
+                Parameters: new[]
+                {
+                    lengthParameter,
+                });
+            ValueExpression contentLengthHeaderExpression = new MultipartFormDataContentExpression(_multipartContentField).Property("Headers").Property("ContentLength");
+            var declaration = new CodeWriterDeclaration("contentLength");
+            VariableReference contentLengthVariable = new VariableReference(typeof(long), declaration);
+            var contentLengthDeclaration = new DeclarationExpression(contentLengthVariable,false);
+            var body = new MethodBodyStatement[]
+            {
+                new IfStatement(BoolExpression.Is(contentLengthHeaderExpression, contentLengthDeclaration))
+                {
+                    Assign(lengthParameter, contentLengthVariable),
+                    Return(Literal(true)),
+                },
```

Comment:
well let us simplify:
```suggestion
                new IfStatement(BoolExpression.Is(contentLengthHeaderExpression, new DeclarationExpression(typeof(long), "contentLength", out var contentLengthVariable)))
                {
                    Assign(lengthParameter, contentLengthVariable),
                    Return(Literal(true)),
                },
```